gtsam/hybrid/DCMixtureFactor.h

```diff
@@ -40,54 +40,64 @@ class DCMixtureFactor : public DCFactor {
 
   /// typedef for DecisionTree which has Keys as node labels and
   /// NonlinearFactorType as leaf nodes.
-  using FactorDecisionTree =
-      DecisionTree<Key, boost::shared_ptr<NonlinearFactorType>>;
+  using Factors = DecisionTree<Key, sharedFactor>;
 
  private:
   /// Decision tree of Gaussian factors indexed by discrete keys.
-  FactorDecisionTree factors_;
+  Factors factors_;
   bool normalized_;
 
  public:
   DCMixtureFactor() = default;
 
+  /**
+   * @brief Construct from Decision tree.
+   *
+   * @param keys Vector of keys for continuous factors.
+   * @param discreteKeys Vector of discrete keys.
+   * @param factors Decision tree with of shared factors.
+   * @param normalized Flag indicating if the factor error is already
+   * normalized.
+   */
   DCMixtureFactor(const KeyVector& keys, const DiscreteKeys& discreteKeys,
-                  const FactorDecisionTree& factors, bool normalized = false)
+                  const Factors& factors, bool normalized = false)
       : Base(keys, discreteKeys), factors_(factors), normalized_(normalized) {}
 
   /**
    * @brief Convenience constructor that generates the underlying factor
    * decision tree for us.
    *
-   * Here it is important that the vector of
-   * factors has the correct number of elements based on the number of discrete
-   * keys and the cardinality of the keys, so that the decision tree is
-   * constructed appropriately.
+   * Here it is important that the vector of factors has the correct number of
+   * elements based on the number of discrete keys and the cardinality of the
+   * keys, so that the decision tree is constructed appropriately.
    *
    * @param keys Vector of keys for continuous factors.
    * @param discreteKeys Vector of discrete keys.
-   * @param factors Vector of factors. Each factor should map to its
-   * corresponding assigned discrete key.
+   * @param factors Vector of shared pointers to factors.
    * @param normalized Flag indicating if the factor error is already
    * normalized.
    */
+  DCMixtureFactor(const KeyVector& keys, const DiscreteKeys& discreteKeys,
+                  const std::vector<sharedFactor>& factors,
+                  bool normalized = false)
+      : DCMixtureFactor(keys, discreteKeys,
+                        Factors(discreteKeys, factors)) {}
+
+  /**
+   * @brief DEPRECATED constructor with non-shared pointers.
+   */
```

Comment:
Add the deprecated guard? Or I can just get rid of it in another PR.

Reply:
Let's discuss roadmap w Kevin first